src/js/repl.js

```diff
@@ -259,34 +275,38 @@ function completer(
 
 export default function startREPL(opts: CLIOptsType): void {
```

Comment:
my comments above make this changes not be needed at all. `repl.js` shouldn't be messed with, I think you can handle reading from stdin in `cljs.js`, i.e. treating the input as a `"text"` script and executing that